source/s3_auto_ranged_put.c

```diff
@@ -174,6 +191,43 @@ static void s_s3_meta_request_auto_ranged_put_destroy(struct aws_s3_meta_request
     aws_mem_release(meta_request->allocator, auto_ranged_put);
 }
 
+static void s_s3_auto_ranged_put_cancel(
+    struct aws_s3_meta_request *meta_request,
+    struct aws_s3_request *failed_request,
+    int error_code) {
+    // TODO: error code needs to be propagated
+    (void)error_code;
+    struct aws_s3_auto_ranged_put *auto_ranged_put = meta_request->impl;
+
+    s_s3_auto_ranged_put_lock_synced_data(auto_ranged_put);
+    if (auto_ranged_put->synced_data.state != AWS_S3_AUTO_RANGED_PUT_STATE_CANCEL &&
```

Comment:
Once this becomes a s_s3_auto_ranged_put_finish function, let's actually just always set the cancel state here, and then call aws_s3_meta_request_push_to_client.